services/docker/docker-helpers.js

```diff
@@ -5,12 +5,42 @@ const dockerBlue = '066da5' // see https://github.com/badges/shields/pull/1690
 function buildDockerUrl(badgeName) {
   return {
     base: `docker/${badgeName}`,
-    pattern: ':user/:repo',
+    pattern: ':user/:repo/:tag*',
```

Comment:
I believe this is going to add the tag to all of the Docker badge routes (haven't checked the others but IIRC they all use this helper). Are we sure that data needed for the other Docker-related badges (build status, pulls, etc.) support filtering by tags?

My guess is that some of them don't, so I suspect this would need to be tweaked a bit. A couple options I can see would be to either conditionally add the trailing `:tag*` route param to the pattern based on a function param (so the service classes that want it/can use it can opt into it), or the handful of service classes that _do_ support working with tags could use this helper to generate the first bits of the pattern and then add what they need on the end.

The latter may be easier if, for example, the tag param would be required for the size badge, etc.

Reply:
You're right that it does add the route param, though not didn't actually affect whether the badge worked or not because we were not doing anything with the tag if it was provided.

I've just fixed this up.

Reply:
Excellent. We want to avoid having an unneeded/unused route param in those other routes to avoid confusing users (they'll be presented with the param in our UI, even though the value isn't used)